### ISSUE TITLE
Keras tensorflow : Freeze calibrated mode, extract graph, input and output tensors

### ISSUE THREAD

**Issue by volvador** [tensorflow/tensorflow#22942]
I create a tensorflow medel using keras embedded in tensorflow:

    import tensorflow as tf
    from tensorflow.python.keras.models import Sequential
    from tensorflow.python.keras.layers import Dense 
    model = Sequential()
    model.add(Dense(input_dim=input_dim, units=hidden_dim))
    model.add(Dense(units=output_dim, activation='linear'))
    model.compile(loss='mse', optimizer='adam')

After fitting the model, I want to freeze it and to extract the graph, the input tensor, and the output tensor. 
    
    session = tf.keras.backend.get_session()
    output_node_names = [node.op.name for node in model.outputs]
    graphdef = tf.graph_util.convert_variables_to_constants(session, session.graph_def, 
    output_node_names)

    with tf.Graph().as_default() as graph:
        tf.import_graph_def(graphdef)
        input_tensor = graph.get_tensor_by_name(graph.get_operations()[0].name+':0')
        output_tensor = graph.get_tensor_by_name(graph.get_operations()[-1].name+':0')

So that after the above code, the variable `graph` will contain the graph, and `input_tensor` and `output_tensor` will contain respectively the  input and output of the graph.

The graph does not offer a method to get its input and output tensors, hence my complicated method above. But it does not work.

Can you help please or advise a better way to do it?

Using Tensorflow 1.5

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by wt-huang** [comment]
@volvador You can also use `session.run()` to retrieve output tensors, just specify the tensor name.
